### ISSUE TITLE
Add a confirmation dialog for moving folders in the file system.

### ISSUE THREAD

**Issue by ettiSurreal** [godotengine/godot-proposals#7862]
### Describe the project you are working on

A simple platformer game, using a git repository to work across multiple devices.

### Describe the problem or limitation you are having in your project

Currently it's **very** easy to accidentally move a file or a folder across the file system, which will often lead to having to search for what and where was moved, and at worst an engine crash. And there is no undo for moving files! I already had two situations where i had to discard an entire commit because i accidentally moved a large folder, which completely messed up references.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a confirmation dialog for moving folders that aren't empty. Individual files are much less likely to be accidentally moved due to a misclick since dragging them on top of eachother does nothing, for that an undo function would work better.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When trying to move a folder that isn't empty into another folder, you get a confirmation dialog, in which you can either cancel the move or confirm it.
I think there should **not** be a "don't show again" checkbox, as i feel like most users would think of it as a first-time warning, while this is supposed to prevent accidental moves due to misinput.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think it can.

### Is there a reason why this should be core and not an add-on in the asset library?

Help users not cause accidental mess-ups.

**Comment by Calinou** [comment]
Given moving a folder can take a long time to update references in a complex project, it makes sense to display a confirmation before this.

**Comment by produno** [comment]
To add to this, it may be a good idea to show in the confirmation box where the folder is being moved to. I have had at least a couple occasions where i thought i was moving a file to one location and it ended up somewhere else.

**Comment by 74boar** [comment]
This is my #1 pain point with the engine. Mistakenly moving the wrong folder can break hundreds of dependencies at once. It should not be possible to suddenly corrupt an entire project by interacting with one of the most basic and frequently used parts of the engine (e.g. navigating the project directory). And even if there was no danger of broken dependencies, I still think a confirmation dialog would be a good QOL feature to prevent moving a folder accidentally. 

Alternatively, a setting to simply disable drag-click on folders would achieve the same goal. Knowing the danger, I never move files with drag-click, I always use right-click -> Move/Duplicate To...

**Comment by nubels** [comment]
I agree that this is a significant usability problem. I think the confirmation dialogue should always be shown, even when moving a single file or folder since it's so easy to accidentally move files using drag and drop. I happens to me frequently and it's always a hassle to resolve.

For now, I've commented out this section of code in `FileSystemDock::can_drop_data_fw` in my fork, relying on the `Move/Duplicate To...` functionality instead.
```cpp
if (drag_data.has("type") && (String(drag_data["type"]) == "files" || String(drag_data["type"]) == "files_and_dirs")) {
	// Move files or dir.
	String to_dir;
	bool favorite;
	_get_drag_target_folder(to_dir, favorite, p_point, p_from);

	if (favorite) {
		return true;
	}

	if (to_dir.is_empty()) {
		return false;
	}

	// Attempting to move a folder into itself will fail later,
	// rather than bring up a message don't try to do it in the first place.
	to_dir = to_dir.ends_with("/") ? to_dir : (to_dir + "/");
	Vector<String> fnames = drag_data["files"];
	for (int i = 0; i < fnames.size(); ++i) {
		if (fnames[i].ends_with("/") && to_dir.begins_with(fnames[i])) {
			return false;
		}
	}

	return true;
}
```
